spec/abstractops.html

```diff
@@ -490,6 +488,24 @@ <h1>ToRelativeTemporalObject ( _options_ )</h1>
     </emu-alg>
   </emu-clause>
 
+  <emu-clause id="sec-temporal-totimezoneidentifier" type="abstract operation">
+    <h1>
+      ToTimeZoneIdentifier (
+        _argument_: an ECMAScript language value,
+      ): either a normal completion containing a String or a throw completion
+    </h1>
+    <dl class="header">
+      <dt>description</dt>
+      <dd>It validates that the argument identifies a built-in offset time zone or built-in named time zone and returns its canonical form.</dd>
+    </dl>
+    <emu-alg>
+      1. Let _S_ be ? ToString(_argument_).
+      1. If ParseText(StringToCodePoints(_S_), |TimeZoneNumericUTCOffset|) is not a List of errors, return _S_.
```

Comment:
It's debatable whether returning _S_ directly, fulfills the requirement for the "canonical form" described in the header, since _S_ could be a string like "-07:00:00.00000" and a "canonical" way of writing that would be "-07:00". (Although not "canonical" in a TZDB sense.)

This canonicalization of offset strings is done in CreateTemporalTimeZone anyway, since all call sites of ToTimeZoneIdentifier pass the result to CreateTemporalTimeZone, so you could consider moving it into here. But I'm not sure if there are any other callers of CreateTemporalTimeZone that rely on it being done there.